src/lib/gundb/UserPropertiesClass.js

```diff
@@ -60,8 +63,11 @@ export default class UserProperties {
 
       //make sure we fetch props first and not having gun return undefined
       await this.props
+
```

Comment:
looks like we have 2 instances of the GUN node (this.gun and this.props). is it ok ? Shouldn't it be as the following:

```javascript
// @flow
import { assign, isUndefined } from 'lodash'
import { defer, from as fromPromise } from 'rxjs'
import { retry } from 'rxjs/operators'

import { REGISTRATION_METHOD_SELF_CUSTODY } from '../constants/login'
import pino from '../logger/pino-logger'

const log = pino.child({ from: 'UserProperties' })

/**
 * Users gundb to handle user storage.
 * User storage is used to keep the user Self Soverign Profile and his blockchain transcation history
 * @class
 *  */
export default class UserProperties {
  /**
   * Default User Properties
   * @type {{isMadeBackup: boolean, firstVisitApp:number, etoroAddCardSpending:boolean, isAddedToHomeScren: false}}
   */
  static defaultProperties = {
    isMadeBackup: false,
    firstVisitApp: null,
    etoroAddCardSpending: true,
    isAddedToHomeScreen: false,
    cameFromW3Site: false,
    lastBonusCheckDate: null,
    countClaim: 0,
    regMethod: REGISTRATION_METHOD_SELF_CUSTODY,
    showQuickActionHint: true,
    registered: false,
    startClaimingAdded: false,
    lastBlock: 0,
  }

  fields = [
    'isMadeBackup',
    'firstVisitApp',
    'etoroAddCardSpending',
    'isAddedToHomeScreen',
    'cameFromW3Site',
    'lastBonusCheckDate',
    'countClaim',
    'regMethod',
    'showQuickActionHint',
    'startClaimingAdded',
  ]

  /**
   * a gun node referring tto gun.user().get('properties')
   * @instance {UserProperties}
   */
  gun: Gun

  data: {}

  constructor(gun: Gun) {
    this.gun = gun.user().get('properties')

    this.ready = (async () => {
      let props
      const { gun, constructor } = this
      const { defaultProperties } = constructor

      // make sure we fetch props first and not having gun return undefined
      await gun

      try {
        props = await defer(() => fromPromise(gun.decrypt())) // decrypt user props
          .pipe(retry(1)) // if exception thrown, retry decrypt one more times
          .toPromise()
      } catch (e) {
        log.error('failed decrypting props', e.message, e)
        props = {}
      }

      if (isUndefined(props) {
        log.warn('undefined props from decrypt')
      }
      
      props = assign({}, defaultProperties, props || {})
      this.data = props

      return props
    })()
  }

  /**
   * Set value to property
   *
   * @param {string} field
   * @param {string} value
   * @returns {Promise<void>}
   */
  async set(field: string, value: any) {
    await this.updateAll({ [field]: value })

    return true
  }

  /**
   * Return property values
   * @param field
   * @returns {Promise<any>}
   */
  get(field: string) {
    return this.data[field]
  }

  /**
   * Return all Properties
   * @returns {{}}
   */
  getAll() {
    return this.data
  }

  /**
   * Set value to multiple properties
   */
  async updateAll(properties: { [string]: any }): Promise<boolean> {
    const { gun, data } = this

    assign(data, properties)
    await gun.secret(data)

    return true
  }

  /**
   * Reset properties to the default state
   */
  async reset() {
    const { gun, constructor } = this
    const { defaultProperties } = constructor

    this.data = assign({}, defaultProperties)
    await gun.secret(defaultProperties)

    return true
  }
}
```

Reply:
fixed